File: 2012m./uzd1./sprendimai/uzd1.py
def getMaxElementIndex(arr):
    maxIndex = 0
    for i in range(1, len(arr)):
        if(arr[maxIndex] < arr[i]):
            maxIndex = i
    return maxIndex

def readData():
    data = open("U1.txt", "r")

    startingPlayers = [None]*5
    startingPlayerIndex = 0
    timeSpentOnBench = [0]*100
    timeSpentPlaying = [0]*100

    for i in range(int(next(data))):
        playerData = next(data).split(" ")
        playerData = [int(o) for o in playerData]
        if(playerData[2] > 0):
            startingPlayers[startingPlayerIndex] = str(playerData[0])
            startingPlayerIndex += 1
        for j in range(2, 2 + playerData[1]):
            if(playerData[j] > 0):
                timeSpentPlaying[playerData[0]] += playerData[j]
            else:
                timeSpentOnBench[playerData[0]] += abs(playerData[j])
    
    data.close()
    
    return startingPlayers, timeSpentOnBench, timeSpentPlaying

def main():
    output = open("U1rez.txt", "w")

    startingPlayers, timeSpentOnBench, timeSpentPlaying = readData()

    mostActivePlayer = getMaxElementIndex(timeSpentPlaying)
    leastActivePlayer = getMaxElementIndex(timeSpentOnBench)

    output.write(f"{' '.join(startingPlayers)}\n")
    output.write(f"{mostActivePlayer} {timeSpentPlaying[mostActivePlayer]}\n")
    output.write(f"{leastActivePlayer} {timeSpentOnBench[leastActivePlayer]}\n")

    output.close()

if __name__ == "__main__":
    main()

File: 2012m./uzd1./sprendimai/uzd1.cpp
#include <iostream>
#include <fstream>
#include <tuple>
#include <algorithm>

using namespace std;

int getMaxElementIndex(const int arr[], int size) {
    int maxIndex = 0;
    for (int i = 1; i < size; i++) {
        if (arr[maxIndex] < arr[i]) {
            maxIndex = i;
        }
    }
    return maxIndex;
}

void readData(string startingPlayers[5], int timeSpentOnBench[100], int timeSpentPlaying[100]) {
    ifstream data("U1.txt");

    int startingPlayerIndex = 0;

    int numPlayers;
    data >> numPlayers;

    for (int i = 0; i < numPlayers; i++) {
        int playerId, numEntries, firstTimeEntry;
        data >> playerId >> numEntries >> firstTimeEntry;

        if (firstTimeEntry > 0) {
            timeSpentPlaying[playerId] += firstTimeEntry;
            startingPlayers[startingPlayerIndex] = to_string(playerId);
            startingPlayerIndex++;
        }
        else {
            timeSpentOnBench[playerId] += abs(firstTimeEntry);
        }

        for (int j = 0; j < numEntries - 1; j++) {
            int timeEntry;
            data >> timeEntry;

            if (timeEntry > 0) {
                timeSpentPlaying[playerId] += timeEntry;
            } else {
                timeSpentOnBench[playerId] += abs(timeEntry);
            }
        }
    }

    data.close();
}

int main() {
    ofstream output("U1rez.txt");

    string startingPlayers[5];
    int timeSpentOnBench[100] = {0};
    int timeSpentPlaying[100] = {0};

    readData(startingPlayers, timeSpentOnBench, timeSpentPlaying);

    int mostActivePlayer = getMaxElementIndex(timeSpentPlaying, 100);
    int leastActivePlayer = getMaxElementIndex(timeSpentOnBench, 100);

    for (const auto& player : startingPlayers) {
        if (!player.empty()) {
            output << player << " ";
        }
    }
    output << "\n";

    output << mostActivePlayer << " " << timeSpentPlaying[mostActivePlayer] << "\n";
    output << leastActivePlayer << " " << timeSpentOnBench[leastActivePlayer] << "\n";

    output.close();

    return 0;
}

